libfs/fs_test.go

```diff
@@ -29,6 +31,44 @@ func makeFS(t *testing.T, subdir string) (
 	return ctx, h, fs
 }
 
+func makeFSWithJournal(t *testing.T, subdir string) (
+	context.Context, *libkbfs.TlfHandle, *FS, func()) {
+	ctx := libkbfs.BackgroundContextWithCancellationDelayer()
+	config := libkbfs.MakeTestConfigOrBustLoggedInWithMode(
+		t, 0, libkbfs.InitSingleOp, "user1")
+
+	tempdir, err := ioutil.TempDir(os.TempDir(), "journal_server")
+	require.NoError(t, err)
+	defer func() {
+		if err != nil {
+			os.RemoveAll(tempdir)
+		}
+	}()
+	err = config.EnableDiskLimiter(tempdir)
+	require.NoError(t, err)
+	err = config.EnableJournaling(
+		ctx, tempdir, libkbfs.TLFJournalSingleOpBackgroundWorkEnabled)
+	require.NoError(t, err)
+	shutdown := func() {
+		libkbfs.CheckConfigAndShutdown(ctx, t, config)
+		err := ioutil.RemoveAll(tempdir)
+		assert.NoError(t, err)
+	}
+
+	h, err := libkbfs.ParseTlfHandle(ctx, config.KBPKI(), "user1", tlf.Private)
+	require.NoError(t, err)
+	fs, err := NewFS(ctx, config, h, subdir, "")
+	require.NoError(t, err)
+
+	// Flush the journal first otherwise tests fail on errNoFlushedRevisions.
```

Comment:
Ah, this code is probably why you didn't have the problem I had in #1191.  This shouldn't need to be here, but we can remove it in #1191 if you want.

Reply:
Hmmm where do you think the fix would be? I think with out this flush, error comes from `SyncFromServerForTesting` called in `Lock`.

Reply:
The fix is 49ddce0.  You can move it to this PR if you want it.

Reply:
Oh I see! I guess to avoid weird conflict it's better to remove this part in #1191 then. Thanks for the fix!